src/aggregate/functions/string.c

```diff
@@ -339,14 +339,20 @@ static int stringfunc_contains(ExprEval *ctx, RSValue *result, RSValue **argv, s
   RSValue *str = RSValue_Dereference(argv[0]);
   RSValue *pref = RSValue_Dereference(argv[1]);
 
-  char *p_str = (char *)RSValue_StringPtrLen(str, NULL);
-  const char *p_pref = (char *)RSValue_StringPtrLen(pref, NULL);
+  size_t p_str_size;
+  char *p_str = (char *)RSValue_StringPtrLen(str, &p_str_size);
+  size_t p_pref_size;
+  const char *p_pref = (char *)RSValue_StringPtrLen(pref, &p_pref_size);
   result->t = RSValue_Number;
 
   size_t num = 0;
-  while ((p_str = strstr(p_str, p_pref)) != NULL) {
-    num++;
-    p_str++;
+  if(p_pref_size > 0) {
+    while ((p_str = strstr(p_str, p_pref)) != NULL) {
+      num++;
+      p_str++;
+    }
+  } else if (p_pref_size == 0) {
+    num = p_str_size + 1;
```

Comment:
Only declare it before
`size_t num;`
And
```suggestion
  if(p_pref_size > 0) {
    num = 0;
    while ((p_str = strstr(p_str, p_pref)) != NULL) {
      num++;
      p_str++;
    }
  } else {
    num = p_str_size + 1;
```